File: xv/patch/XV2020FX.swift

extension XV2020 {
  
  enum FX {
    static let patchWerk = XV.FX.patchWerk(params: parms.params())

    static let parms = XV3080.FX.parms + [
      .p([.type], 0x00, .options(fxTypeOptions)),
      .p([.out], 0x04, .options(outOptions)),
    ]
    
    static let fxTypeOptions: [Int:String] = 41.dict {
      [$0 : XV.FX.fxDisplayName($0)]
    }

    static let outOptions: [Int:String] = OptionsParam.makeOptions(["A"])
  }

  enum Chorus {
    static let patchWerk = XV.Chorus.patchWerk(params: parms.params())

    static let parms = XV5050.Chorus.parms + [
      .p([.type], 0x00, .opts(["Off", "Chorus"])),
      .p([.out, .assign], 0x02, .options(FX.outOptions)),
    ]
  }

  enum Reverb {
    static let patchWerk = XV.Reverb.patchWerk(params: parms.params())

    static let parms = XV5050.Reverb.parms + [
      .p([.type], 0x00, .opts(["Off", "Reverb"])),
      .p([.out, .assign], 0x02, .options(FX.outOptions)),
    ]

  }
}

File: xv/xv2020_fx.js

const fxTypeOptions: [Int:String] = 41.dict {
  [$0 : XV.FX.fxDisplayName($0)]
}

const outOptions = ["A"]

const parms = XV3080FX.parms.concat([
  ['type', { b: 0x00, opts: fxTypeOptions }],
  ['out', { b: 0x04, opts: outOptions }],
])

const chorusParms = XV5050FX.chorusParms.concat([
  ['type', { b: 0x00, opts: ["Off", "Chorus"] }],
  ['out/assign', { b: 0x02, opts: outOptions }],
])

const reverbParms = XV5050FX.reverbParms.concat([
  ['type', { b: 0x00, opts: ["Off", "Reverb"] }],
  ['out/assign', { b: 0x02, opts: outOptions }],
])

module.exports = {
  patchWerk: XVFX.patchWerk(parms),
  chorusPatchWerk: XVFX.chorusPatchWerk(chorusParms),
  reverbPatchWerk: XVFX.reverbPatchWerk(reverbParms),
}

